docs/spec.yml

```diff
@@ -299,11 +299,261 @@ paths:
+          description: Unauthorised Access
+        '404':
+          description: No resource found which matches identifiers
+  '/v3/technical-records/letter/{systemNumber}/{createdTimestamp}':
+    post:
+      summary: 'Generates a letter for a technical record'
+      tags:
+        - Generate Letter
+      parameters:
+        - in: path
+          name: systemNumber
+          schema:
+            type: string
+          required: true
+          description: This represents the systemNumber of the vehicle. Partition key of the DynamoDB table.
+        - in: path
+          name: createdTimestamp
+          schema:
+            type: string
+          required: true
+          description: Datetime at which the record was created. Sort key of the DynamoDB table.
+      operationId: letter
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object 
+              required:
+                - vtmUsername
+                - recipientEmailAddress
+                - letterType
+                - paragraphId
+              properties:
+                vtmUsername:
+                  type: string
+                recipientEmailAddress:
+                  type: string
+                letterType:
+                  type: string
+                  enum:
+                    - 'trailer acceptance'
+                    - 'trailer rejection'
+                paragraphId:
+                  type: number
+                  enum:
+                   - 3
+                   - 4
+                   - 5
+                   - 6
+                   - 7
+      responses:
+        '200':
+          description: Success
+        '401':
+          description: Not authenticated
+        '403':
+          description: Unauthorised Access
+        '404':
+          description: No resource found which matches identifiers
+  '/v3/technical-records/promote/{systemNumber}/{createdTimestamp}':
+    post:
+      summary: 'Promotes the specified technical record to current'
+      tags:
+        - Promote a Technical Record
+      parameters:
+        - in: path
+          name: systemNumber
+          schema:
+            type: string
+          required: true
+          description: This represents the systemNumber of the vehicle. Partition key of the DynamoDB table.
+        - in: path
+          name: createdTimestamp
+          schema:
+            type: string
+          required: true
+          description: Datetime at which the record was created. Sort key of the DynamoDB table.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object 
+              required:
+                - reasonForPromoting
+              properties:
+                reasonForPromoting:
+                  type: string
+      responses:
+        '200':
+          description: Returns the promoted technical record
```

Comment:
complete / subset

Reply:
as it can be either, do we need to specify? Seems to make the language more clunky to me